src/huggingface_hub/_upload_large_folder.py

```diff
@@ -199,6 +202,18 @@ def __init__(self, items: List[JOB_ITEM_T]):
             else:
                 logger.debug(f"Skipping file {paths.path_in_repo} (already uploaded and committed)")
 
+    def target_chunk(self) -> int:
+        with self._chunk_lock:
+            return COMMIT_SIZE_SCALE[self._chunk_idx]
+
+    def update_chunk(self, success: bool, duration: float) -> None:
+        with self._chunk_lock:
+            if not success and self._chunk_idx > 0:
+                self._chunk_idx -= 1
+            else:
+                if duration < 40 and self._chunk_idx < len(COMMIT_SIZE_SCALE) - 1:
+                    self._chunk_idx += 1
```

Comment:
3 suggestions here:
- I would increase the number of files per commit only if the previous commit was already full. Otherwise we might increase the limit only because a commit had a few files in it. I added a `nb_items: int` to check this.
- I would clamp the value between 0 and `len(COMMIT_SIZE_SCALE) - 1` at the end of the method. By doing this if we extend the policy at some point we don't have to care about these details since it's handled once for every case.
- I would add a log when we increase/decrease commit size

```suggestion
    def update_chunk(self, success: bool, nb_items: int, duration: float) -> None:
        with self._chunk_lock:
            if not success:
                logger.warn(f"Failed to commit {nb_items} files at once. Will retry with less files in next batch.")
                self._chunk_idx -= 1
            elif nb_items >= COMMIT_SIZE_SCALE[self._chunk_idx] and duration < 40:
                logger.info(f"Successfully committed {nb_items} at once. Increasing the limit for next batch.")
                self._chunk_idx += 1

            self._chunk_idx = max(0, min(self._chunk_idx, len(COMMIT_SIZE_SCALE) - 1))
```